bleak/backends/bluezdbus/client.py

```diff
@@ -426,8 +426,13 @@ async def _cleanup_all(self) -> None:
         Free all the allocated resource in DBus. Use this method to
         eventually cleanup all otherwise leaked resources.
         """
-        await self._remove_signal_handlers()
-        self._disconnect_message_bus()
+        async with self._cleanup_lock:
+            await self._remove_signal_handlers()
+            self._disconnect_message_bus()
+
+            # Reset all stored services.
+            self.services = BleakGATTServiceCollection()
+            self._services_resolved = False
```

Comment:
I moved this here since a disconnect occurring from an event should clean out services too. Not only when it's a manual event.

Reply:
Nice. :+1:

Reply:
Nice. :+1: